src/t_lpm.lua

```diff
@@ -23,3 +24,44 @@ assert(ret.val == "test")
 
 ret = acl:lookup(lpm.tobin("10.2.2.1"))
 assert(ret == nil)
+
+
+-- test overwrite/remove
+local tracer = require("tracer")
```

Comment:
Why not just write the tracer in Lua?  You can implement `__gc` metamethods there too. It could probably be a nearly one-liner lambda function to modify a global variable.  Do I miss something?

Reply:
Annoyingly Lua 5.1 and earlier versions of LuaJIT don't support __gc on anything that isn't userdata.

From: http://lua-users.org/wiki/MetatableEvents
>  __gc - Userdata finalizer code. When userdata is set to be garbage collected, if the metatable has a __gc field pointing to a function, that function is first invoked, passing the userdata to it. The __gc metamethod is not called for tables. (See http://lua-users.org/lists/lua-l/2006-11/msg00508.html)

I suppose I could have replaced __gc in the metatable of some other bit of userdata (even liblpm's userdata) but that would look rather confusing and might be sensitive to changes in liblpm, which is less than ideal IMO.

    tracer = lpm.new()

Reply:
Indeed, that is pretty annoying.. however, there is a `newproxy()` hack and I still prefer it over a tracer in C.
```lua
function prepare_for_gc()
  if _VERSION == "Lua 5.1" then
    local tracer = newproxy(true)
    getmetatable(tracer).__gc = function () tracer_gc = true end
  else
    local tracer = {}
    setmetatable(tracer, {__gc = function () tracer_gc = true end})
  end
  tracer_gc = false -- global
  return tracer
end

tracer = prepare_for_gc()
collectgarbage()
tracer = nil
collectgarbage()
assert(tracer_gc == true)
```
Please use this abstraction.  It works with Lua 5.1 and 5.2 for me.

Reply:
Done (with an interface identical to the C module).

Note: there's a bug in the `prepare_for_gc` function above, the tracer is defined as local in the wrong scope, so the tracer is GC'd "immediately" after leaving the `if ... end` block. 😸

Reply:
Hah!  Glad somebody has an actually working brain.